ktor-shared/ktor-shared-serialization-kotlinx/common/src/io/ktor/shared/serialization/kotlinx/KotlinxSerializationConverter.kt

```diff
@@ -49,7 +60,10 @@ public class KotlinxSerializationConverter(private val format: SerialFormat = De
             return result
         }
 
-        val guessedSearchSerializer = guessSerializer(value, format.serializersModule)
+        val matchedSerializer = extraSerializerMatchers.entries.firstOrNull { (_, matcher) ->
```

Comment:
Am I right that this is used only for `JsonElement`? If yes, can you test that serialization works without this code? Then we can delete it and simplify the logic here. My guess that it was required for some old version of the library, but now it's redundant. 
I have some other concerns about it, but lets see if we can just delete it first.

Reply:
Yes and no. At the moment it's only an escape hatch for `json()` and is not used by `cbor()`. I can check if `json()` works without it when receiving JsonObject. Will report back.

Reply:
Sorry for the delay - had a busy week. I finally got a chance to test JsonElement serialization without `match` API and it works just fine. Removed that API altogether now.